client/sites/tools/logs/hooks/use-actions.tsx

```diff
@@ -0,0 +1,218 @@
+	const siteGsmOffsetDisplay =
+		siteGmtOffset === 0 ? 'UTC' : `UTC${ siteGmtOffset > 0 ? '+' : '' }${ siteGmtOffset }`;
+	const locale = useSelector( getCurrentUserLocale );
+	const getFormattedDate = useMemo(
+		() => ( value: string ) => {
+			const dateFormat = locale === 'en' ? 'll [at] h:mm A' : 'h:mm A, ll';
+			const formattedDate = moment( value )
+				.utcOffset( siteGmtOffset * 60 )
+				.format( dateFormat );
+			return <span>{ formattedDate }</span>;
+		},
+		[ locale, siteGmtOffset ]
+	);
+
+	const actions = useMemo( () => {
+		if ( logType === 'php' ) {
+			return [
+				{
+					id: 'copy-msg',
+					label: translate( 'Copy message' ),
+					icon: copy,
+					isPrimary: true,
+					disabled: isLoading,
+					supportsBulk: false,
+					callback: async ( items: ( PHPLog | ServerLog )[] ) => {
+						const message = ( items[ 0 ] as PHPLog ).message;
+						try {
+							await navigator.clipboard.writeText( message );
+							dispatch(
+								successNotice(
+									/* translators: notice shown upon copy of Logs entry */
+									translate( 'Copied' )
+								)
+							);
+						} catch ( error ) {
+							dispatch(
+								errorNotice(
+									/* translators: notice shown upon failed copy of Logs entry */
+									translate( 'Copy failed' )
+								)
+							);
+						}
+					},
+				},
+				{
+					id: 'details-modal',
+					label: translate( 'View log details' ),
+					modalHeader: translate( 'Log Details' ),
+					isPrimary: false,
+					disabled: isLoading,
+					supportsBulk: false,
+					RenderModal: ( { items }: { items: PHPLog[] } ) => {
+						const item = items[ 0 ] as PHPLog;
+						return (
+							<div className="site-logs-details-modal">
+								<div className="site-logs-details-modal__field-title">
+									{
+										// translators: %s is the timezone offset of the site, e.g. GMT, GMT +1, GMT -1.
+										translate( 'Date & time (%(siteGsmOffsetDisplays):', {
+											args: { siteGsmOffsetDisplay },
+										} )
+									}
+								</div>
+								<div>{ getFormattedDate( item.timestamp ) }</div>
+								<div className="site-logs-details-modal__field-title">
+									{ translate( 'Severity:' ) }
+								</div>
+								<div>
+									<Badge className={ `badge--${ item.severity }` }>{ item.severity }</Badge>
+								</div>
+								<div className="site-logs-details-modal__field-title">
+									{ translate( 'Message:' ) }
+								</div>
+								<div>{ item.message }</div>
+								<div className="site-logs-details-modal__field-title">{ translate( 'Kind:' ) }</div>
+								<div>{ item.kind }</div>
+								<div className="site-logs-details-modal__field-title">{ translate( 'Name:' ) }</div>
+								<div>{ item.name }</div>
+								<div className="site-logs-details-modal__field-title">{ translate( 'File:' ) }</div>
+								<div>{ item.file }</div>
+								<div className="site-logs-details-modal__field-title">{ translate( 'Line:' ) }</div>
+								<div>{ item.line }</div>
```

Comment:
This looks like it's a number we are rendering in the HTML. Do we expect this to go over 1000? Would it make sense to parse with `numberFormat` so it gets properly localised? e.g. have the correct thousandth separators (`1,000` for EN, `1.000` for EL/DE, etc.)

Reply:
That's a good thought. I've used `numberFormat` to formatting this and `body_bytes_sent`.